src/engine/strat_engine/pool.rs

```diff
@@ -269,12 +269,10 @@ impl StratPool {
         let mut needs_save = metadata.thinpool_dev.fs_limit.is_none()
             || metadata.thinpool_dev.feature_args.is_none();
 
-        if action_avail != ActionAvailability::NoPoolChanges {
-            needs_save |= match thinpool.check(uuid, &mut backstore) {
-                Ok(check) => check.0,
-                Err(e) => return Err((e, backstore.into_bdas())),
-            };
-        }
+        needs_save |= match thinpool.check(uuid, &mut backstore) {
```

Comment:
For background here of why we decided to put this in an if condition:

1. A pool with inconsistent block sizes was created using a previous version of stratisd.
2. On set up, current version of stratisd detects and puts in NoPoolChanges mode.
3. Then check method is run and, because new version of stratisd has greedy allocation algorithm for thinpool metadata device, the metadata device is extended. This makes needs_save true.
4. Then, because needs_save is true, write_metadata is called, and pool setup returns an error.

The thing we were trying to avoid, then, is having the pool setup encounter what we thought of an an unnecessary error. It was also the case that after the expanson of the thin meta device, the DM devices were inconsistent w/ the pool metadata, and that couldn't be fixed w/out a stratisd restart.

Another way the pool could encounter an error on setup is if the ```started``` status has changed, or needs to be updated. I believe we could have encountered the error in our previous testing using inconsistent block sizes. I may try to formulate another test around that.

My proposal, though, which eliminates the code here entirely is to simply remove the two places where we call ```ThinPool::check``` outside of separate event threads, as it is my belief that both calls were introduced before the timer thread was introduced, and that the introduction of the timer thread resolves the problems that these calls were expected to handle.

That's in https://github.com/stratis-storage/stratisd/pull/3223, which I'm putting up for your review.

Reply:
Resolved.